extensions/widgets/segment_new/segment_new.lcb

```diff
@@ -0,0 +1,716 @@
+/*
+Copyright (C) 2015 Runtime Revolution Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+/*
+Description: This widget is a segmented control.
+A segmented control is a horizontal control that is made up of multiple segments, where each segment functions as a discrete button.
+*/
+
+-- declaring extension as widget, followed by identifier
+widget com.livecode.extensions.livecode.segmented
+--
+
+-- adding metadata to ensure the extension displays correctly in livecode
+metadata title is "Segmented"
+metadata author is "Georgia Hutchings"
+metadata version is "1.0.0"
+--
+
+-- dependency declarations
+use com.livecode.canvas
+use com.livecode.widget
+use com.livecode.engine
+--
+
+-- property declarations
+property theme				get mTheme				set setTheme
+property segmentStyle 		get mStyle				set setStyle			-- string (enum) : segment or bar
```

Comment:
It might be best if this property were just called 'style', then it is consistent with the style property of classic controls.